File: solutions/training-java/src/main/java/org/example/Algorithms/Astar/Node.java
package org.example.Algorithms.Astar;

public class Node {
    private int x;
    private int y;

    private double g;

    private double h;

    private double f;

    private Node parent;

    ////////////////////////////////////////////////////////////////////
    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public double getG() {
        return g;
    }

    public double getF() {
        return f;
    }

    public Node getParent() {
        return parent;
    }

    public void setParent(Node n){
        parent = n;
    }

    ////////////////////////////////////////////////////////////////////

    public Node(int x, int y){
        this.x = x;
        this.y = y;
    }

    public void setNode(Node n){
        if(AstarAlg.maze[this.x][this.y] == "0"){
            this.g = 1;
        }else{
            this.g = 100;
        }

        this.h = euclideanDistance(n);

        this.f = this.g + this.h;
    }

    public double euclideanDistance(Node n){
        return Math.sqrt(Math.pow((n.x-this.x),2) + Math.pow((n.y-this.y), 2));
    }

    @Override
    public String toString(){
        return "["+this.x+"-"+this.y+"] --> G="+this.g +"/ H="+this.h +"/ F="+this.f +"/";
    }


    @Override
    public boolean equals(Object obj) {
        return this.x == ((Node) obj).x && this.y == ((Node) obj).y;
    }
}

File: solutions/training-kotlin/src/main/kotlin/Algorithms/ASTAR/Node.kt
package Algorithms.ASTAR

class Node {
    var x = 0
    var y = 0

    var G = 0.0
    var H = 0.0
    var F = 0.0

    var parent: Node? = null

    constructor(x: Int, y: Int) {
        this.x = x
        this.y = y
    }

    fun setNode(node: Node) {
        if (Algorithm.maze.get(x).get(y) == "0") G = 1.0 else G = 100.0
        H = euclideanDistance(node)
        F = G + H
    }

    fun euclideanDistance(node: Node): Double = Math.sqrt(Math.pow((node.x - x).toDouble(), 2.0) + Math.pow((node.y - y).toDouble(), 2.0))

    override fun toString(): String {
        return "[ $x - $y ] --> G= $G/ H= $H/ F= $F/"
    }

    override fun equals(other: Any?): Boolean {
        other as Node
        return if (this.x == other.x && this.y == other.y) return true else return false
    }
}
